src/Provider/Provider.Web/Orchestrators/Part1/LocationOrchestrator.cs

```diff
@@ -117,64 +135,64 @@ public async Task<VacancyEmployerInfoModel> GetVacancyEmployerInfoModelAsync(Vac
             }
             else
             {
-                selectedOrganisation = 
-                    employerVacancyInfo.LegalEntities.Single(l => l.LegalEntityId == vacancy.LegalEntityId);
+                selectedOrganisation =
+                    employerVacancyInfo.LegalEntities.Single(l => l.LegalEntityId == vacancy.LegalEntityId);                
             }
 
-            if (m.UseOtherLocation == true)
-            {
-                vacancy.EmployerLocation = new Vacancies.Client.Domain.Entities.Address
-                {
-                    AddressLine1 = m.AddressLine1,
-                    AddressLine2 = m.AddressLine2,
-                    AddressLine3 = m.AddressLine3,
-                    AddressLine4 = m.AddressLine4,
-                    Postcode = m.Postcode.AsPostcode()
-                };
-            }
-            else
+            long? selectedOrganisationId = selectedOrganisation.LegalEntityId;
+
+            var legalEntityId = selectedOrganisationId.HasValue ? selectedOrganisationId : vacancy.LegalEntityId;
+
+            var employerProfile =
+                await _recruitVacancyClient.GetEmployerProfileAsync(vacancy.EmployerAccountId, legalEntityId.GetValueOrDefault());
+
+            var allLocations = await GetAllAvailableLocationsAsync(employerProfile, vacancy, ukprn);
+
+            var newLocation =
+                locationEditModel.SelectedLocation == LocationViewModel.UseOtherLocationConst
+                    ? locationEditModel.ToAddressString()
+                    : locationEditModel.SelectedLocation;
+
+            var matchingAddress = GetMatchingAddress(newLocation, allLocations);
+
+            vacancy.EmployerLocation = matchingAddress != null ? matchingAddress : ConvertToDomainAddress(locationEditModel);
+
+            //if cookie is found then update legal entity and name option from cookie
+            if (employerInfoModel != null)
             {
-                vacancy.EmployerLocation = new Vacancies.Client.Domain.Entities.Address
-                {
-                    AddressLine1 = selectedOrganisation.Address.AddressLine1,
-                    AddressLine2 = selectedOrganisation.Address.AddressLine2,
-                    AddressLine3 = selectedOrganisation.Address.AddressLine3,
-                    AddressLine4 = selectedOrganisation.Address.AddressLine4,
-                    Postcode = selectedOrganisation.Address.Postcode.AsPostcode()
-                };
+                vacancy.LegalEntityName = selectedOrganisation.Name;
```

Comment:
This is happening twice. I would prefer this is exactly same as Employer, which has gone through enough testing and it is optimized.

Reply:
it was copy paste error!